### ISSUE TITLE
Field props onChange accepts event or value, but onChange callback only accepts events (not plain values)

### ISSUE THREAD

**Issue by naw** [erikras/redux-form#3446]
According to the docs, it's valid to call `onChange` with _either_ a synthetic event or a value. My form component calls it with a simple value (like a string or number).

However, if an `onChange` _callback_ is defined on the given `Field`, then redux-form assumes the value is an event object and tries to extend it with a `preventDefault` key. This doesn't work for values like strings and numbers.

This is the code I'm referring to: https://github.com/erikras/redux-form/blob/6e262c2945fd93c6f0b06441487deaba27e8cef0/src/ConnectedField.js#L113-L119

I'm not sure if the code needs to be updated or the docs need to be updated. I'd be happy to work on a PR, if you let me know what the correct resolution is. Thank you.

**Comment by ChengYurou** [comment]
i have same question

**Comment by nadezhda-devlabs** [comment]
The same for me. I'm using **react-select** and its onChange prop returns an object `{ label, value }`, instead of the event. **redux-form** require us to pass an event to its `input.onChange` function.

So I hot fixed it, as adding a new custom prop`onChanged` to the Field, in order to get the correct value, when the Select is changed:

```
import React from 'react'
import ReactSelect from 'react-select'

const Select = ({value, name, onChange, onChanged, type, disabled, className = 'c-custom-select',
  placeholder, options, clearable, multi, optionRenderer}) => {
  return <ReactSelect
    // ... all needed props
    onChange={selected => {
      onChange(selected ? selected.value : null)

      if (onChanged) onChanged(selected ? selected.value : null)
    }}
    />
}

export default Select
```

However, it will be great if `onChange` accepts both - event and a value.

**Comment by stephane-ruhlmann** [comment]
I had to do the same hotfix, not very convenient...

**Comment by arvinsim** [comment]
Is this already fixed? I have the same problem where the `onChange` handler's arguments changes on some calls.

**Comment by erikras** [comment]
Published fix in [`v8.0.0`](https://github.com/erikras/redux-form/releases/tag/v8.0.0).
